Fig.Tests/TypedSettingsTests.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace Fig.Test
+{
+    public class TypedSettingsTests
+    {
+        private Settings _settings;
+
+        [SetUp]
+        public void Setup()
+        {
+            this._settings = new SettingsBuilder()
+                .UseSettingsDictionary(new SettingsDictionary() {
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Name)}"] = "Fullname",
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Age)}"] = "40",
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Pi)}"] = "3.14"
+                })
+                .Build();
+        }
+
+        [Test]
+        public void WhenBindReturnsValue()
+        {
+            var testClass = this._settings.Bind<TestClass>();
+
+            Assert.AreEqual("Fullname", testClass.Name);
+            Assert.AreEqual(40, testClass.Age);
+            Assert.AreEqual(3.14, testClass.Pi);
+        }
+
+        [Test]
+        public void WhenBindAndMissingSettingsPropertyThrowsException()
+        {
+            this._settings = new SettingsBuilder()
+                .UseSettingsDictionary(new SettingsDictionary() {
+                    [$"{nameof(TestClass)}.{nameof(TestClass.Name)}"] = "Fullname"
+                })
+                .Build();
+
+            TestClass testClass;
+
+            Assert.Throws<KeyNotFoundException>(() => testClass = this._settings.Bind<TestClass>());
```

Comment:
Do we need to take the return value? I would guess not and in that case we could get rid of line 42

Reply:
True.  I can take that out.